### PR TITLE
Refactor glvis main file; fix for segfault on glvis script

### PR DESCRIPTION
**Pull request by kanye-quest**
- Try to remove some duplicated code in glvis.cpp
- Split up `StartVisualization` function into `GlvisInitVis` and `GlvisStartVis` functions; this allows calling additional window/scene functions before entering the main loop
- Fix `PlayScript` segfault by setting ScriptControl function after window created in `GlvisInitVis()`

### CODE REVIEW

**Comment by v-dobrev** [comment]
One issue I noticed: if `GlvisInitVis` fails, then `GlvisStartVis` is still run but it should not be.

The logic in the removed portion of the code was very similar to, but not exactly the same as, the code that is now reused. Hopefully, this is not going to create new issues.

File: glvis.cpp [hunk 1]
```diff
@@ -334,31 +335,25 @@ int ReadStream(istream &is, const string &data_type)
...
-   const char *win_title =
-      (window_title == string_default) ? window_titles[t] : window_title;
-
-   return InitVisualization(win_title, window_x, window_y,
-                            window_w, window_h);
-}
 
 // Visualize the data in the global variables mesh, sol/grid_f, etc
-void StartVisualization(int field_type)
+void GlvisInitVis(int field_type)
```
Comment (tzanio) [inline glvis.cpp:340]: For consistency, can we rename `Glvis_` -> `GLVis_`?

Comment (tzanio) [reply glvis.cpp]: Done in 9436a0e656bc82b3b6f2a70f5d0409530297557d, revert if you don't like :)

**Review by tzanio — APPROVED** [review]
Worked in all the test I've tried, thanks @kanye-quest !

**Review by v-dobrev — APPROVED** [review]

**Comment by v-dobrev** [comment]
I didn't test it myself (as @tzanio did) but it looks good. 👍

**Pull request merged**